Remove buildtools

sbomer opened the pull request:
I think this is finally ready to go. @dotnet/coreclr-infra this removes the dependency on buildtools. Please let me know if you think there's any additional testing I need to do.
- Makes a number of common test projects SDK-style, and replaces the buildtools prerelease restore target with the restore logic built in to the SDK. For some projects, we currently restore 5.0 assets that are used for netcoreapp3.0 - this required a workaround to switch the TFM in a few cases.
- Replaces the corefx testhost deps file creation logic with that from the SDK's publish logic.
- Uses a new IL sdk (as the old one used to make many redundant file copies), and uses an .ilproj to restore ilasm/ildasm up-front.
- Uses the UpdateVersions logic from arcade instead of buildtools.

This opens up the possibility of further simplifying some of our msbuild imports, but I don't want to do so in this PR as it's already a sizeable change.

sbomer committed:
Start removing buildtools

sbomer committed:
Clean up project files and remove dependency on prerelease restore

Instead we use the built-in SDK targets to restore packages. The new
logic will not copy pdbs from package dependencies to CORE_ROOT.

sbomer committed:
Don't generate deps.json in TargetingPack, and remove depproj targets

sbomer committed:
Remove some more references to buildtools

sbomer committed:
Remove references to binclashlogger

sbomer committed:
Avoid using GenerateDepsJson task for corefx testhost

sbomer committed:
Fix order of core_root projects

sbomer committed:
Don't define Build target in dir.targets

hoyosjs commented on dependencies.props:
```diff
@@ -17,7 +17,6 @@
   <!-- Source of truth for dependency tooling: the commit hash of the dotnet/versions master branch as of the last auto-upgrade. -->
   <PropertyGroup>
     <CoreClrCurrentRef>5d3c9a7c54c1c59b764de0e2dfb6bbb4ce29476c</CoreClrCurrentRef>
```
This was mostly used to update ILAsmVersion.txt. I see you are removing that file, should we just remove the MicrosoftNETCoreRuntimeCoreCLRPackageVersion update and make a DarcMaestro subscription for the CoreCLR packages? Also, if that happens, should we delete this file and move anything interesting left to Directory.Build.props? Are there scenarios that depend on this as it currently is?

sbomer replied:
Eventual goal is to get rid of this file - I'll look into it and may work on that in this change if it makes sense.

sbomer committed:
Make CoreCLRTestLibrary SDK-style

sbomer committed:
Fix test targeting pack build

sbomer committed:
Remove duplicate imports in testwrapper

sbomer committed:
Filter external.depproj to prevent conflicts

sbomer committed:
Avoid using dotnet on path for batch restore

sbomer committed:
Restore CoreCLRTestLibrary

sbomer committed:
Restore ilasm/ildasm for .ilproj tests

sbomer committed:
Use recent IL SDK

This prevents issues with the old SDK trying to copy ilasm to the
Tools directory for each build of an ilproj.

sbomer committed:
Fix targeting pack build

Use netcoreapp3.0 to ensure that package dependencies show up in the
build output (bin/TargetingPack). This fixes a missing reference for
CscBench, which depends on Microsoft.CodeAnalysis.CSharp.

Also clean up some unnecessary properties.

sbomer committed:
Remove some unused properties

sbomer committed:
Update performance projects

sbomer committed:
Don't error on package downgrade

sbomer committed:
Rename CoreFX.depproj and clean up

sbomer committed:
Fix coreclrtestlibrary casing

sbomer committed:
Don't check for Tools directory in stress dependencies

sbomer committed:
Restore corefx before generating testhost

sbomer committed:
Disable warnings from the SDK's restore logic

sbomer committed:
Set ilasm RID

sbomer committed:
Fix corefx testhost restore

sbomer committed:
Override IL SDK rids for musl and rhel

sbomer committed:
Support building netcoreapp5.0 test projects with 3.0 SDK

sbomer committed:
Fix external.depproj warnings

sbomer committed:
Add back NuGetTargetMoniker for test dependencies

This lets tests build against the netcoreapp5.0 assets file created
for test_dependencies.csproj.

sbomer committed:
Fix warning in external.csproj build

This project is built to lay out assets, but was giving a warning
because no source files were specified.

sbomer committed:
Fix netcoreapp version mismatch for CoreCLRTestLibrary

This library needs to be restored for netcoreapp5.0, and built for
netcoreapp3.0 to match the tests.

sbomer committed:
Fix NuGetTargetMoniker for performance projects

sbomer committed:
Fix restore TFM for test_dependencies and test_runtime

sbomer committed:
Switch TFM for CoreFX testhost and prevent error in external.csproj

sbomer committed:
Clean up comments

sbomer committed:
Fix NuGetTargetMoniker for test wrappers

sbomer committed:
Restore external.csproj up-front

sbomer committed:
Target 3.0 for generated testhost deps.json

There were problems when trying to target 5.0 using the current SDK,
which only supports targeting 3.0 out of the box.

This required using a more recent SDK, as p6 restored framework
references differently. p7 is able to restore a framework reference
with the MicrosoftNETCoreApp version that we desire.

sbomer committed:
Use UpdatePublishedVersions from arcade

sbomer committed:
Cleanup and add better comments

sbomer commented:
/azp run coreclr-outerloop

sbomer committed:
Fix RID for rhel

sbomer committed:
Ensure featuredefines are imported for test library

sbomer commented:
/azp run coreclr-outerloop

jashook reviewed (approved):
lgtm thank you for this change!

hoyosjs commented on eng/updateversions.proj:
```diff
@@ -0,0 +1,11 @@
+
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('Directory.Build.props', '$(MSBuildThisFileDirectory)../'))" />
+
+  <Import Project="$(NuGetPackageRoot)\microsoft.dotnet.versiontools.tasks\$(MicrosoftDotNetVersionToolsTasksPackageVersion)\build\Microsoft.DotNet.VersionTools.Tasks.targets" />
```
Technically this is the only thing that we still have from buildtools?

sbomer replied:
This now comes from arcade: https://github.com/dotnet/arcade/tree/master/src/Microsoft.DotNet.VersionTools :)

echesakovMSFT commented on tests/dir.props:
```diff
@@ -59,14 +31,6 @@
     <AltJitArch>$(__AltJitArch)</AltJitArch>
```
We can git of this in the future.

echesakovMSFT commented on tests/src/Common/Directory.Build.targets:
```diff
@@ -1,8 +1,39 @@
+        <File>%(Identity)</File>
+      </AllResolvedRuntimeDependencies>
+      <RunTimeDependecyCopyLocalFile Include="@(AllResolvedRuntimeDependencies)"  Exclude="@(RunTimeDependecyExcludeFiles)"/>
+      <RunTimeDependecyCopyLocal Include="@(RunTimeDependecyCopyLocalFile -> '%(File)')"  />
```
We can fix the typo in the future.

echesakovMSFT commented on tests/src/Common/Directory.Build.targets:
```diff
@@ -1,8 +1,39 @@
+    </ItemGroup>
+
+    <Copy
+      SourceFiles="@(RunTimeDependecyCopyLocal)"
```
Typo

echesakov reviewed (approved):
Looks Great! Thank you!
